bio_pieces/ctleptop.py

```diff
@@ -0,0 +1,289 @@
+    nearbyPermutations("AAA")
+    set(['AAA'])
+    """
+    if (index >= len(letters)):
+        return set([''])
+    subWords = nearbyPermutations(letters, index + 1)
+    nearbyLetters = getNearbyChars(letters[index])
+    return permutations(subWords, nearbyLetters)
+
+
+def permutations(subWords, nearbyLetters):
+    """(set, list) -> (set)
+    >>>permutations(set(['CA']), ['A', 'T'])
+    set(['ACA', 'TCA'])
+    """
+    permutations = set()
+    for subWord in subWords:
+        for letter in nearbyLetters:
+            permutations.add(letter + subWord)
+    return permutations
+
+
+def getaalist(codonlist):
+    """(list) -> (list)
+    Return aa list from a a given nt codon list.
+    >>>getaalist(['AAA','ACT'])
+    ['K', 'T']
+    """
+    aalist = []
+    for codon in codonlist:
+        aa = Seq(codon, IUPAC.unambiguous_dna)
+        aa = str(translate(aa))
+        aalist.append(aa)
+    return aalist
+
+
+def list_overlap(list1, list2):
+    """(str, list) -> bool
+    Return True  if the two list hava element that overlaps.
+
+    >>>list_overlap('RAC',['B', 'D', 'H', 'K', 'M', 'N', 'S', 'R', 'W', 'V', 'Y'])
+    True
+    >>>list_overlap('ACT',['B', 'D', 'H', 'K', 'M', 'N', 'S', 'R', 'W', 'V', 'Y'])
+    False
+
+    """
+    for i in list1:
+        if i in list2:
+            return True
+    return False
+
+
+# define our method
+#def replace_all(text, dic):
+    #"""(str, dict)-> (str)
+    #>>>replace_all()
+    #"""
+    #for i, j in dic.iteritems():
+        #text = text.replace(i, j)
+    #return text
+
+
+def access_mixed_aa(file_name):
+    """(str) ->(list,list,list,list).
+    Return a list of amino acide code for ambiguous dna codon, position of
+    ambiguous nt codon, aa name,seq id from fasta header  by reading multifasta
+    nucleotide fasta file
+    """
+    from Bio import SeqIO
+    aa = []
+    nucleotide_idx = []
+    nucl_codon = []
+    seqids = []
+    for seq_record in SeqIO.parse(file_name, 'fasta'):
+        seq_id = seq_record.id
+        seq_len = len(seq_record)
+        header, seqline = seq_record.id, str(seq_record.seq)
+    # for header, seqline in readFasta(file_name):
+        # print header + "\n" + seq_line
+
+        # my_seq = Seq(seq_line, IUPAC.extended_dna)
+        my_seq = Seq(str(seqline), IUPAC.ambiguous_dna)
+        # seq2 = Seq("ARAWTAGKAMTA", IUPAC.ambiguous_dna)
+        # seq2 = seq2.translate()
+        # print seq2
+        # print ambiguous_dna_values["W"]
+        # print IUPAC.ambiguous_dna.letters
+        seqline = seqline.replace("-", "N")
+        n = 3
+        codon_list = {i + n: seqline[i:i + n] for i in range(0, len(seqline), n)}
+        # print yaml.dump(ambi_codon)
+        # print yaml.dump(codon_list)
+        ambi_nucl = AMBICODON.keys()
+        # print ambi_nucl
+        # print ambi_codon["Y"]
+        for key, codon in sorted(codon_list.items()):
+            # print "key: ", key , "codon:", codon
+            if list_overlap(codon, ambi_nucl):
```

Comment:
It seems that there is some redundancy here

I think this can replace the if statement and should be quite a bit faster than the current implementation
```
items = set(m).intersection(ambi_nucl)
if items:
```